clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/AbstractKafkaConfigProperties.java

```diff
@@ -67,17 +91,30 @@ protected final void setSpecificClientConfig(final Map<String, String> specificC
     }
 
     /**
-     * Sets the prefix for the client ID that is passed to the Kafka server to allow application specific server-side
-     * request logging.
+     * Sets the component name to be included in the client ID along with a UUID in case no component unique ID
+     * has been set via {@link #setComponentUid(String)} and also no unique ID could be determined automatically
+     * because this application is not running in Kubernetes.
+     *
+     * @param componentName The component name to use.
+     * @throws NullPointerException if the component name is {@code null}.
+     */
+    public final void setComponentName(final String componentName) {
```

Comment:
My understanding is that this method should be called in code. It is not meant to map configuration properties (e.g. from Spring Boot). Maybe renaming it to a name that does not start with "set" would help here.

Reply:
It wouldn't make much sense for a user to set this property via a configuration file, yes. 
But on the other hand, it wouldn't necessarily be a problem and we wouldn't put this property in the list of supported configuration properties in the documentation anyway. (And it can only be set using Spring which be want to move away from.)
So, I would rather keep this standard naming style.

Reply:
You are technically right. On the other hand, it is not only about Spring. The naming suggests that the class has a property `componentName` that can be changed in the configuration. FMPOV this naming has caused a lot of confusion in our discussions in the last couple of weeks. If I recall correctly, @sophokles73 has explicitly asked for more clarity.

Reply:
> The naming suggests that the class has a property `componentName` that can be changed in the configuration.

I see no problem per se in the `AbstractKafkaConfigProperties` having different kinds of properties.
There are the properties in the map given to the Kafka client constructor (no individual property setters there).
There may also be additional properties for configuring the Hono Kafka client classes (like `pollTimeout`).
And now we configure properties used for the client ID creation.

We could make the meaning of the new methods more explicit by renaming the methods to `setComponentNameForClientId` / `setComponentUIdForClientId`. WDYT?

Reply:
> And now we configure properties used for the client ID creation.

FMPOV the confusion arises from the fact that `componentName` is not a _configuration property_. It is not intended to be configured externally by providing a property in a configuration file, or an environment variable, or a Java system property. It is a method that you use to provide a value to this class. Usually, this is prefixed by "set". But this is not a random Java class but a specific one, namely one that implements configuration properties. And a method starting by "set" on this kind of class suggests that there is a property with that name that is intended to be configured with one of the mechanisms mentioned above. 

The fact that we are having this discussion is proof that it is not just another innocent setter. It was exactly this naming issue that first led to us accidentally setting the client ID by configuration in the IoT Packages Project, then even led to this bug being considered a feature. Then even effort was invested to implement this anti-feature for Quarkus as well. @sophokles73 WDYT?

Reply:
I tend to agree with @b-abel here. 
Why do we need to set the component name on the properties object in the first place? Where is the component name being used/required?

Reply:
> Why do we need to set the component name on the properties object in the first place? Where is the component name being used/required?

In the code before this PR, the client ID created in `AbstractKafkaConfigProperties` contained a "(default)ClientIdPrefix". As value for that prefix, the component name (e.g. "Hono AMQP Adapter") was set.
This means, when looking at client IDs in the Kafka broker logs (or e.g. Confluent UI), it was being made clear in which Hono component the client was operating.

With #2906, the intention was to improve on this by using the Kubernetes Pod/container IDs instead.

Now, with this PR, I've kept the "(default)ClientIdPrefix" but renamed it to "componentName(ForClientId)".
As stated in the "setComponentNameForClientId" javadoc, this name is used for the created client ID if the component isn't running in Kubernetes, i.e. if the Kubernetes Pod/container IDs can't be used in the client ID.

So, in practice, "setComponentNameForClientId" could also be removed if we want to ignore Docker (ie. integration test) or non-K8s scenarios. That would also be OK with me.

> And a method starting by "set" on this kind of class suggests that there is a property with that name that is intended to be configured with one of the mechanisms mentioned above.

For the Quarkus components this is not the case since we have separate `[*]Options` classes there (e.g. `KafkaConsumerOptions`).

Reply:
> So, in practice, "setComponentNameForClientId" could also be removed if we want to ignore Docker (ie. integration test) or non-K8s scenarios. That would also be OK with me.

I've removed it now.
`setComponentUidForClientId` has been renamed to `overrideComponentUidUsedForClientId` - it is only used in unit tests here.

Reply:
> setComponentUidForClientId has been renamed to overrideComponentUidUsedForClientId - it is only used in unit tests here.

I like the name. But I'm not sure what the purpose of this method is now. Is it intended to be invoked by productive code in the future? Or only for unit tests?

Reply:
I've extended the javadoc now:
`This method is mainly intended for usage in non-Kubernetes environments and unit tests.`